### PR TITLE
FLEX-5165 Add datalake optimization policy template

### PR DESCRIPTION
**Pull request by midir99**
### Description
Initial implementation of a Policy (Azure Data Lake Optimization) which enables customers to run efficiently for Azure Data Lake. The policy type is Usage Reduction and provide the recommendations with Potential savings.
 
### Issues Resolved
https://flexera.atlassian.net/browse/FLEX-5165
 
### Link to Example Applied Policy
https://app.flexera.com/orgs/34088/automation/applied-policies/projects/135557?policyId=6761dfcc0f8449d761662eff
 
### Contribution Check List
- [ ] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable
- [X] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

**Review by XOmniverse — CHANGES_REQUESTED** [review]

File: cost/azure/data_lake_optimization/data_lake_optimization.pt [hunk 1]
```diff
@@ -0,0 +1,1157 @@
+name "Azure Data Lake Optimization"
+rs_pt_ver 20180301
+type "policy"
+short_description "Checks Azure Data Lake storage accounts for old blobs and move the said blobs to the Cool or Archive tier after user approval. See the [README](https://github.com/flexera-public/policy_templates/tree/master/cost/azure/data_lake_optimization) and [docs.flexera.com/flexera/EN/Automation](https://docs.flexera.com/flexera/EN/Automation/AutomationGS.htm) to learn more."
+long_description ""
+severity "low"
+category "Cost"
+default_frequency "weekly"
+info(
+  version: "1.0.0",
```
[the remark this answers is not in the document]
Comment (midir99) [reply cost/azure/data_lake_optimization/data_lake_optimization.pt]: thanks @XOmniverse

File: cost/azure/data_lake_optimization/data_lake_optimization.pt [hunk 2]
```diff
@@ -0,0 +1,1157 @@
...
+
+datasource "ds_azure_storage_accounts_tag_filtered" do
+  run_script $js_azure_storage_accounts_tag_filtered, $ds_azure_storage_accounts, $param_exclusion_tags, $param_exclusion_tags_boolean
+end
+
+script "js_azure_storage_accounts_tag_filtered", type: "javascript" do
+  parameters "ds_azure_storage_accounts", "param_exclusion_tags", "param_exclusion_tags_boolean"
+  result "result"
+  code <<-EOS
+
```
[the remark this answers is not in the document]
Comment (midir99) [reply cost/azure/data_lake_optimization/data_lake_optimization.pt]: thanks @XOmniverse, this change was addressed

File: cost/azure/data_lake_optimization/data_lake_optimization.pt [hunk 3]
```diff
@@ -0,0 +1,1157 @@
...
+    }
+  })
+EOS
+end
+
+datasource "ds_azure_distinct_regions" do
+  run_script $js_azure_distinct_regions, $ds_azure_blobs_with_tier
+end
+
+script "js_azure_distinct_regions", type: "javascript" do
```
[the remark this answers is not in the document]
Comment (midir99) [reply cost/azure/data_lake_optimization/data_lake_optimization.pt:627]: thanks @XOmniverse, this change was addressed

**Review by night-codes — DISMISSED** [review]

**Review by XOmniverse — CHANGES_REQUESTED** [review]
Requested a few more changes after reviewing again

File: cost/azure/data_lake_optimization/data_lake_optimization.pt [hunk 4]
```diff
@@ -0,0 +1,1140 @@
...
+  label "Automatic Actions"
+  description "When this value is set, this policy will automatically take the selected action."
+  allowed_values ["Update Blobs Storage Tier", "Delete Blobs"]
+  default []
+end
+
+parameter "param_hierarchical_namespace" do
+  type "string"
+  category "Filters"
+  label "Apply filtering for the Azure Data Lake Storage Accounts"
```
[the remark this answers is not in the document]
Comment (midir99) [reply cost/azure/data_lake_optimization/data_lake_optimization.pt]: Hey @Shawn, thank you for the review, I changed this, let me know if new version is ok.

File: cost/azure/data_lake_optimization/data_lake_optimization.pt [hunk 5]
```diff
@@ -0,0 +1,1140 @@
...
+  description "Apply filtering for the Azure Data Lake Storage Accounts where the hierarchical namespace is enabled"
```
[the remark this answers is not in the document]
Comment (midir99) [reply cost/azure/data_lake_optimization/data_lake_optimization.pt]: Hey @Shawn, thank you for the review, I changed this, let me know if new version is ok.

File: cost/azure/data_lake_optimization/data_lake_optimization.pt [hunk 6]
```diff
@@ -0,0 +1,1140 @@
...
+end
+
+script "js_azure_storage_accounts_hns", type: "javascript" do
+  parameters "ds_azure_storage_accounts_region_filtered", "param_hierarchical_namespace"
+  result "result"
+  code <<-EOS
+  if (param_hierarchical_namespace === "Yes") {
+    result = _.filter(ds_azure_storage_accounts_region_filtered, function(account) {
+      //The value will be undefined for the accounts which are only Storage Accounts
+      if(account['hnsEnabled']){
```
[the remark this answers is not in the document]
Comment (midir99) [reply cost/azure/data_lake_optimization/data_lake_optimization.pt]: Hey @Shawn, thank you for the review, I changed this, let me know if new version is ok.

**Review by XOmniverse — APPROVED** [review]
LGTM

**Pull request merged**